tests/application_development/cpp/testcase.yaml

```diff
@@ -0,0 +1,3 @@
+tests:
+  test:
+    tags: cpp
```

Comment:
if this is going to be a compile test only, i.e. this test is not supposed to run, then please add:

build_only: true

Also, AFAIK c++ is still not supported with native_posix, so you want to add arch_exclude: posix as well.